File: backend/app/schemas/deficit.py
# app/schemas/deficit.py
from typing import List, Optional
from pydantic import BaseModel


class DeficitDayEntry(BaseModel):
    date: str
    day: int
    intake: float
    walking_burn: float
    burn: float
    deficit: float
    cumulative_deficit: float
    steps: int
    tracked: bool


class DeficitMonthlyResponse(BaseModel):
    year: int
    month: int
    weight_kg: Optional[float] = None
    baseline: float = 2000.0
    kcal_per_kg: float = 7700.0
    tracked_days: int
    total_intake: float
    total_burn: float
    net_deficit: float
    estimated_weight_change_kg: float
    days: List[DeficitDayEntry]

File: frontend/src/models/deficit.ts
// src/models/deficit.ts
export interface DeficitDayEntry {
  date: string;
  day: number;
  intake: number;
  walking_burn: number;
  burn: number;
  deficit: number;
  cumulative_deficit: number;
  steps: number;
  tracked: boolean;
}

export interface DeficitMonthlyResponse {
  year: number;
  month: number;
  weight_kg?: number | null;
  baseline: number;
  kcal_per_kg: number;
  tracked_days: number;
  total_intake: number;
  total_burn: number;
  net_deficit: number;
  estimated_weight_change_kg: number;
  days: DeficitDayEntry[];
}
